### PR TITLE
setColorPriseSchema

### PR DESCRIPTION
**Pull request by conanvster**

### CODE REVIEW

File: client/app/admin/price-schema/price-schema.controller.js [hunk 1]
```diff
@@ -10,9 +10,19 @@ export default class PriceSchemaController {
...
     this.priceSchemaId = '';
+    this.colors = [
+      {color: '#8b54aa', colorName: 'violet'},
+      {color: '#ffcc00', colorName: 'yellow'},
+      {color: '#6f89c0', colorName: 'blue'},
+      {color: '#54aa6a', colorName: 'green'},
+      {color: '#ff972f', colorName: 'orange'}
+    ];
+
   }
```
Comment (DmytroLypovetskiy) [inline client/app/admin/price-schema/price-schema.controller.js]: just a question
Why do you keep colors in js, not in css?

File: client/app/admin/price-schema/price-schema.html [hunk 2]
```diff
@@ -42,6 +43,12 @@
                     {{$ctrl.currentTribune.name}}
                 </td>
                 <td>
+                    <span class=" fa fa-2x fa-square" style="color: {{$ctrl.currentTribune.color}}"></span>
```
Comment (DmytroLypovetskiy) [inline client/app/admin/price-schema/price-schema.html]: inline styles are bad practice

File: client/app/admin/price-schema/price-schema.html [hunk 3]
```diff
@@ -42,6 +43,12 @@
...
+                    <select class="form-control" name="color" id="color" ng-change="$ctrl.currentTribune.color" ng-model="$ctrl.currentTribune.color">
+                        <option ng-repeat="color in $ctrl.colors" value="{{color.color}}">{{color.colorName}}</option>
+                    </select>
+                </td>
+                <td>
                     <div class="form-group"  ng-class="{'has-error': priceForm.tribunePrice.$invalid && priceForm.tribunePrice.$dirty}">
```
Comment (DmytroLypovetskiy) [inline client/app/admin/price-schema/price-schema.html]: just a question 
should it be only single `{`?

File: client/app/sector/sector.html [hunk 4]
```diff
@@ -70,7 +70,7 @@
       <p>Лучшие места для просмотра матчей команды «Металлист 1925»!</p>
 
       <p>Корпоративные ложи Skybox-современные и комфортабельные ложи для VIP-персон, вместимостью от 11 до 20 мест, расположенные на 4 уровне Западной трибуны стадиона «Металлист».</p>
-      <p>Стоимость аренды одного места на одном матче - <strong>800 грн.</strong></p>
+      <p>Стоимость аренды одного места на одном матче - <strong>1000 грн.</strong></p>
       <p></p>К вашим услугам :</p>
```
Comment (DmytroLypovetskiy) [inline client/app/sector/sector.html]: wrong escaped paragraph

File: client/app/stadium/stadium.controller.js [hunk 5]
```diff
@@ -38,24 +38,7 @@ export default class StadiumController {
...
-
-      return this.getColorByPrice(price);
-    }
-  }
-
-  getColorByPrice(price) {
-    return this.colors
-      .filter(color => color.price == price)
-      .map(color => color.color)[0];
+        return this.getPriceBySector(tribuneName, sectorNumber, this.priceSchema);
```
Comment (DmytroLypovetskiy) [inline client/app/stadium/stadium.controller.js]: why getColor method returns getPriceBySector? does it return color?

File: client/app/stadium/stadium.controller.js [hunk 6]
```diff
@@ -66,16 +49,20 @@ export default class StadiumController {
   getPriceBySector(tribuneName, sectorNumber, priceSchema) {
```
Comment (DmytroLypovetskiy) [inline client/app/stadium/stadium.controller.js]: incorrect name for method. It returns color so better name is `getColorBySector`

File: client/app/stadium/stadium.controller.js [hunk 7]
```diff
@@ -66,16 +49,20 @@ export default class StadiumController {
...
+      }
     } else {
-      if (!priceSchema['tribune_' + tribuneName]['sector_' + sectorNumber].price) {
-        return priceSchema['tribune_' + tribuneName].price;
+      if (!priceSchema['tribune_' + tribuneName]['sector_' + sectorNumber].color) {
+        return priceSchema['tribune_' + tribuneName].color;
       }
-      return priceSchema['tribune_' + tribuneName]['sector_' + sectorNumber].price;
+      return priceSchema['tribune_' + tribuneName]['sector_' + sectorNumber].color;
     }
```
Comment (DmytroLypovetskiy) [inline client/app/stadium/stadium.controller.js]: DRY - Do Not Repeat Yourself
You wrote `priceSchema['tribune_' + tribuneName]` 7 times. So it should be `const yourVariable = priceSchema['tribune_' + tribuneName];`

than your if 
`
if (!priceSchema['tribune_' + tribuneName].color) {
        return '#808080';
       } else {
         return priceSchema['tribune_' + tribuneName].color;
       }
`
should be
`
return (!yourVariable.color) ? '#808080' : yourVariable.color;
`

or even better
`
return yourVariable.color || '#808080';
`

And save `'#808080'` to variable like `const defaultColor = '#808080';` and if you will need to update this color, you will do it only in one place

File: client/app/sector/sector.html (hunk 4 above)
Comment (DmytroLypovetskiy) [inline client/app/sector/sector.html]: And better to save price or prices to variables

File: client/app/admin/price-schema/color-schema-editor/color-schema-editor.component.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      if ( changes.currentPriceSchema ) {
+        this.priceSchema = Object.assign({}, this.currentPriceSchema);
+        this.colors = Object.assign({}, this.colors);
+      }
+    }
+
+    edit(colorPrice) {
+      this.onEdit({
+        $event: {
+          colorPrice: colorPrice
```
Comment (Skadabr) [inline client/app/admin/price-schema/color-schema-editor/color-schema-editor.component.js]: bad name

File: client/app/admin/price-schema/color-schema-editor/color-schema-editor.component.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
+import template from './color-schema-editor.html';
+
+let colorSchemaEditorComponent = {
+  bindings: {
+    currentPriceSchema: '<',
```
Comment (Skadabr) [inline client/app/admin/price-schema/color-schema-editor/color-schema-editor.component.js]: why are you passing currentPriceSchema and colors when you can just use colors

**Pull request merged**